sklearn/utils/estimator_checks.py

```diff
@@ -149,6 +150,32 @@ def check_estimator_sparse_data(name, Estimator):
         raise
 
 
+def check_dtype_object(name, Estimator):
```

Comment:
I'd appreciate a comment here just to give a sense of what this requires of an estimator: `Numeric features must be accepted when dtype=object` for example.